integrations/bitbucket-cloud/bitbucket_cloud/helpers/file_kind_live_event.py

```diff
@@ -0,0 +1,238 @@
+import asyncio
+from pathlib import Path
+from initialize_client import init_client
+from typing import Any, Optional
+import json
+import yaml
+from loguru import logger
+from bitbucket_cloud.client import BitbucketClient
+from fnmatch import fnmatch
+
+
+FILE_PROPERTY_PREFIX = "file://"
+JSON_FILE_SUFFIX = ".json"
+YAML_FILE_SUFFIX = (".yaml", ".yml")
+
+
+def extract_hash_from_payload(changes: dict[str, Any]) -> tuple[str, str, str]:
+    new_hash = changes["new"]["target"]["hash"]
+    old_hash = changes["old"]["target"]["hash"]
+    branch = changes["new"]["name"]
+    return new_hash, old_hash, branch
+
+
+def determine_action(diff_stat: dict[str, Any]) -> tuple[str, str, str]:
+    old = diff_stat.get("old", {})
+    new = diff_stat.get("new", {})
+    old_file_path = old["path"] if old else ""
+    new_file_path = new["path"] if new else ""
+    if not old:
+        return "added", old_file_path, new_file_path
+    elif not new:
+        return "deleted", old_file_path, new_file_path
+    return "modified", old_file_path, new_file_path
+
+
+async def process_file_value(
+    value: str,
+    parent_directory: str,
+    repository: str,
+    hash: str,
+    client: BitbucketClient,
+) -> Any:
+    if not isinstance(value, str) or not value.startswith(FILE_PROPERTY_PREFIX):
+        return value
+
+    file_meta = Path(value.replace(FILE_PROPERTY_PREFIX, ""))
+    file_path = f"{parent_directory}/{file_meta}"
+    bitbucket_file = await client.get_repository_files(repository, hash, file_path)
+
+    return (
+        parse_file(bitbucket_file, file_path)
+        if file_path.endswith(JSON_FILE_SUFFIX)
+        else bitbucket_file
+    )
+
+
+async def process_dict_items(
+    data: dict[str, Any],
+    parent_directory: str,
+    repository: str,
+    hash: str,
+    client: BitbucketClient,
+    diff_stat: Optional[dict[str, Any]] = None,
+    repo: Optional[dict[str, Any]] = None,
+    branch: Optional[str] = None,
+) -> dict[str, Any]:
+    tasks = [
+        process_file_value(value, parent_directory, repository, hash, client)
+        for value in data.values()
+    ]
+    processed_values = await asyncio.gather(*tasks)
+
+    result = dict(zip(data.keys(), processed_values))
+    if diff_stat and repo and branch:
+        result = {
+            "content": result,
+            "metadata": diff_stat,
+            "repo": repo,
+            "branch": branch,
+        }
+    return result
+
+
+async def process_list_items(
+    data: list[dict[str, Any]],
```

Comment:
should be a type "FileObject" or something like that as its hard to understand what is expected there